squire_lib/src/accounts.rs

```diff
@@ -63,19 +76,33 @@ pub struct OrganizationAccount {
 }
 
 impl SquireAccount {
-    /// Creates a new account object
-    pub fn new(user_name: String, display_name: String) -> Self {
-        Self {
-            user_name,
+    pub fn create(user_name: String, display_name: String, permissions: SharingPermissions) -> Self {
+        SquireAccount {
             display_name,
-            user_id: Uuid::new_v4().into(),
-            arena_name: None,
-            mtgo_name: None,
-            trice_name: None,
-            do_share: SharingPermissions::Everything,
+            user_name,
+            gamer_tags: HashMap::new(),
+            user_id: UserAccountID::new(Uuid::new_v4()),
+            permissions
+        }
+    }
+
+    pub fn update_tags(&mut self, platfrom: Platforms, user_name: String) {
+        self.gamer_tags.insert(platfrom, user_name);
+    }
+
+    pub fn get_tags(&self, platforms: Vec<Platforms>) -> Vec<Option<String>> {
```

Comment:
I'm curious as to why this is your design. We should definitely have a method, `.get_tag`, for getting one tag. I also see the use in having a method that returns a copy of all the stored tags. However, if I have a `Vec` of platforms, I can do something like this. I see this as being a fairly niche use case, though.
```rust
let tags: Vec<Option<String>> = platforms.iter().map(|p| account.get_tag(p)).collect();
```

Reply:
I just figured having a list would allow for more flexibility, but if you think that wouldn't be worth it then I can easily change it.